File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {

        int lSubString = 0;
        int subString = 0;
        Set<Character> chars = new HashSet<>();
        for (int i = 0; i < s.length(); i++){
            chars.add(s.charAt(i));
            subString++;
            for (int j = i +1; j<s.length(); j++) {
                if (chars.contains(s.charAt(j))) {
                    break;
                }
                chars.add(s.charAt(j));
                subString++;
            }
            lSubString = Math.max(lSubString, subString);
            subString = 0;
            chars.clear();
        }
        return lSubString;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_set = set()

        left = 0

        maximum_length = 0

        for right in range(len(s)):
            while s[right] in char_set:
                char_set.remove(s[left])
                left+=1

            char_set.add(s[right])

            maximum_length = max(maximum_length, right-left+1)

        return maximum_length